pwiz_tools/Skyline/TestFunctional/PasteMoleculesTest.cs

```diff
@@ -1772,6 +1773,39 @@ void TestFileImportTransitionList(string knownGood)
             //OkDialog(messageDlg, messageDlg.AcceptButton.PerformClick); // Acknowledge the error
         }
 
+        // Test for an issue where a mass-only description of a labeled precursor would come up with the unlabeled mass for the precursor transition
+        void TestHeavyPrecursorNoFormulas()
+        {
+            var input = 
+                "Molecule List Name,Precursor Name, Precursor m/z,Precursor Charge, Product m/z,Product Charge, Label Type,Explicit Retention Time,Explicit Retention Time Window\n" +
+                "molecules1,2′deoxycitidine,330.1095,-1,330.1095,-1,light,7.7,2\n" +
+                "molecules1,2′deoxycitidine,336.12963,-1,336.12963,-1,heavy,7.7,2\n";
+            var asFile = false;
+            for (var pass = 0; pass < 2; pass++)

```

Comment:
I probably would have written this loop as:
`foreach (bool asFile in new[] {false, true})`